futures-util/src/select.rs

```diff
@@ -0,0 +1,38 @@
+//! The `select` macro.
+
+#[macro_export]
+macro_rules! select {
+    ($(
+        $name:ident => $body:expr,
+    )*) => { {
+        $(
+            $crate::assert_unpin(&$name);
+        )*
+        loop {
```

Comment:
I'm going to attempt to change this loop to use `MaybeDone` like `join!` does so that `$body` can go outside the loop, making `break` work like you'd expect.

Reply:
Done.